src/components/Form/Form.tsx

```diff
@@ -0,0 +1,132 @@
+import * as PropTypes from 'prop-types'
+import * as React from 'react'
+import * as _ from 'lodash'
+
+import { UIComponent, childrenExist, customPropTypes } from '../../lib'
+import { ComponentVariablesInput, ComponentPartStyle } from '../../../types/theme'
+import {
+  ComponentEventHandler,
+  Extendable,
+  ReactChildren,
+  ShorthandValue,
+  ShorthandRenderFunction,
+} from '../../../types/utils'
+import FormField from './FormField'
+
+export interface IFormProps {
+  action?: string
+  as?: any
+  children?: ReactChildren
+  className?: string
+  content?: ShorthandValue
+  onSubmit?: ComponentEventHandler<IFormProps>
+  fields?: ShorthandValue[]
+  renderField?: ShorthandRenderFunction
+  styles?: ComponentPartStyle
+  variables?: ComponentVariablesInput
+}
+
+/**
+ * A Form displays a set of related user input fields in a structured way.
+ */
+class Form extends UIComponent<Extendable<IFormProps>, any> {
+  static create: Function
+
+  public static displayName = 'Form'
+
+  public static className = 'ui-form'
+
+  public static propTypes = {
+    /** An element type to render as (string or function). */
+    as: customPropTypes.as,
+
+    /**
+     *  Form content for childrenApi
+     *  @docSiteIgnore
+     */
+    children: PropTypes.node,
+
+    /** Additional CSS class name(s) to apply.  */
+    className: PropTypes.string,
+
+    /** Shorthand for primary content. */
+    content: customPropTypes.contentShorthand,
+
+    /** Shorthand array of props for the Form.Fields inside the Form. */
+    fields: customPropTypes.collectionShorthand,
+
+    /**
+     * A custom render iterator for rendering each of the Form fields.
+     * The default component, props, and children are available for each field.
+     *
+     * @param {React.ReactType} Component - The computed component for this slot.
+     * @param {object} props - The computed props for this slot.
+     * @param {ReactNode|ReactNodeArray} children - The computed children for this slot.
+     */
+    renderField: PropTypes.func,
+
+    /**
+     * The HTML form submit handler.
+     * @param {SyntheticEvent} event - React's original SyntheticEvent.
+     * @param {object} data - All props.
+     */
+    onSubmit: PropTypes.func,
+
+    /** The HTML form action. */
+    action: PropTypes.string,
+
+    /** Additional CSS styles to apply to the component instance.  */
+    styles: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+
+    /** Override for theme site variables to allow modifications of component styling via themes. */
+    variables: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+  }
+
+  public static defaultProps = {
+    as: 'form',
+  }
+
+  static Field = FormField
```

Comment:
let's keep consistency within the file at least when it comes to modifiers; if we explicitly use `public` let's add it to every public member

Reply:
@mnajdova not addressed

Reply:
@mnajdova please disregard; Github is not that good at showing the correct changes..it was actually addressed